hyperspy/drawing/marker.py

```diff
@@ -0,0 +1,123 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with  Hyperspy.  If not, see <http://www.gnu.org/licenses/>.
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+
+class MarkerBase(object):
+
+    """Marker that can be added to the signal figure
+
+    Attributes
+    ----------
+    marker_properties : dictionary
+        Accepts a dictionary of valid (i.e. recognized by mpl.plot)
+        containing valid line properties. In addition it understands
+        the keyword `type` that can take the following values:
+        {'line', 'text'}
+    """
+
+    def __init__(self):
+        # Data attributes
+        self.data = None
+        self.axes_manager = None
+        self.ax = None
+        self.auto_update = True
+
+        # Properties
+        self.marker = None
+        self._marker_properties = {}
+
+    @property
+    def marker_properties(self):
+        return self._marker_properties
+
+    @marker_properties.setter
+    def marker_properties(self, kwargs):
+
+        for key, item in kwargs.iteritems():
+            if item is None and key in self._marker_properties:
+                del self._marker_properties[key]
+            else:
+                self._marker_properties[key] = item
+        if self.marker is not None:
+            plt.setp(self.marker, **self.marker_properties)            
+            try:
+                # self.ax.figure.canvas.draw()
+                self.ax.hspy_fig._draw_animated()
+            except:
+                pass
+
+    def set_marker_properties(self, **kwargs):
+        """
+        Set the line_properties attribute using keyword
+        arguments.
+        """
+        self.marker_properties = kwargs
+
+    def set_data(self, x1=None, y1=None,
+                 x2=None, y2=None, text=None, size=None):
+        """
+        Set data to the structured arra. Each field of data should have
+        the same dimensions than the nagivation axes. The other fields are
+        overwritten.
+        """
+        if np.alltrue([hasattr(a, "__iter__") is False
+                      for a in [x1, y1, x2, y2, text, size]]):
+            self.auto_update = False
+        self.data = np.array((np.array(x1), np.array(y1),
+                              np.array(x2), np.array(y2),
+                              np.array(text), np.array(size)),
+                             dtype=[('x1', object), ('y1', object),
+                                    ('x2', object), ('y2', object),
+                                    ('text', object), ('size', object)])
+
+    def add_data(self, **kwargs):
+        """
+        Add data to the structured array. Each field of data should have
+        the same dimensions than the nagivation axes. The other fields are
+        not changed.
+        """
+        if np.alltrue([hasattr(kwargs[key], "__iter__") is False
+                      for key in kwargs.keys()]):
+            self.auto_update = False
```

Comment:
At the moment the `self.auto_update` is set to `False` if you add only float data. If, however, for the "untouched" fields you had arrays, the `self.auto_update` should still be left as `True`. I suggest moving the check to after "merging" the `kwargs` with `self.data` (and obviously changing that it checks `self.data` keys, and not `kwargs` )